src/linker/Linker.Steps/ResolveFromXmlStep.cs

```diff
@@ -86,17 +89,26 @@ protected override void Process ()
 				if (Context.IgnoreDescriptors)
 					return;
 			}
+
+			if (!ShouldProcessElement (nav))
+				return;
+
 			try {
-				ProcessAssemblies (Context, nav.SelectChildren ("assembly", _ns));
+				ProcessAssemblies (nav.SelectChildren ("assembly", _ns));
 			} catch (Exception ex) when (!(ex is LinkerFatalErrorException)) {
 				throw new LinkerFatalErrorException (MessageContainer.CreateErrorMessage ($"Failed to process description file '{_xmlDocumentLocation}'", 1004), ex);
 			}
 		}
 
-		protected virtual void ProcessAssemblies (LinkContext context, XPathNodeIterator iterator)
+		protected virtual void ProcessAssemblies (XPathNodeIterator iterator)
 		{
 			while (iterator.MoveNext ()) {
-				AssemblyDefinition assembly = GetAssembly (context, GetAssemblyName (iterator.Current));
+				var name = GetAssemblyName (iterator.Current);
```

Comment:
Nit: I don't understand why we need to call GetAssemblyName before checking for feature switches... it will work, just weird that you're changing it...

Reply:
Parsing a null/empty assemblyname in AssemblyNameReference.Parse can throw an exception - I wanted that error not to be hidden by the feature switches.

Reply:
OK - maybe add a comment so that it's not moved later on.